File: 2017-javascript/day02.js
import { importInput } from "./helpers.js";
const input = importInput("./inputs/2")
  .split("\n")
  .map(line => line.split(/\s+/));
input.pop();

// Part 1
let partOne = 0;
for (let nums of input) {
  let max = Math.max(...nums);
  let min = Math.min(...nums);
  partOne += max - min;
}
// Part 2
let partTwo = 0;
for (let nums of input) {
  for (let i = 0; i < nums.length; i++) {
    for (let j = 0; j < nums.length; j++) {
      if (i !== j && nums[i] % nums[j] === 0) {
        partTwo += nums[i] / nums[j];
      }
    }
  }
}
console.log("===== Day 2 =====");
console.log("Part One: ", partOne);
console.log("Part Two: ", partTwo);

File: 2020-python/day02.py
input = open('./inputs/2', 'r').read().split('\n')
input.pop()


class Password:
    def __init__(self, input: str):
        inputString = input.split(': ')
        self.password: str = inputString[1]
        self.policy: list[str] = inputString[0].split(' ')
        self.letter: str = self.policy[1]
        self.min: int = int(self.policy[0].split('-')[0])
        self.max: int = int(self.policy[0].split('-')[1])

    def isValid(self) -> bool:
        return self.min <= self.password.count(self.letter) <= self.max

    def isValid2(self) -> bool:
        return (self.password[self.min-1] == self.letter) != (self.password[self.max-1] == self.letter)


# part one
count1 = 0
for i in range(len(input)):
    pw: Password = Password(input[i])
    if pw.isValid():
        count1 += 1
print(count1)

# part two
count2 = 0
for i in range(len(input)):
    pw: Password = Password(input[i])
    if pw.isValid2():
        count2 += 1
print(count2)
